File: sample/src/main/java/me/xiaopan/assemblyadaptersample/itemfactory/GameChildItemFactory.java
package me.xiaopan.assemblyadaptersample.itemfactory;

import android.content.Context;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import me.xiaopan.assemblyadapter.AssemblyChildItem;
import me.xiaopan.assemblyadapter.AssemblyChildItemFactory;
import me.xiaopan.assemblyadaptersample.R;
import me.xiaopan.assemblyadaptersample.bean.Game;

public class GameChildItemFactory extends AssemblyChildItemFactory<GameChildItemFactory.GameChildItem> {

    private EventListener eventListener;

    public GameChildItemFactory(Context context) {
        this.eventListener = new EventProcessor(context);
    }

    @Override
    public boolean isTarget(Object itemObject) {
        return itemObject instanceof Game;
    }

    @Override
    public GameChildItem createAssemblyItem(ViewGroup parent) {
        return new GameChildItem(R.layout.list_item_game, parent);
    }

    public interface EventListener{
        void onClickIcon(int position, Game user);
        void onClickName(int position, Game user);
        void onClickLike(int position, Game user);
    }

    private static class EventProcessor implements EventListener {
        private Context context;

        public EventProcessor(Context context) {
            this.context = context;
        }

        @Override
        public void onClickIcon(int position, Game game) {
            Toast.makeText(context, "瞅这游戏这臭逼样！", Toast.LENGTH_SHORT).show();
        }

        @Override
        public void onClickName(int position, Game game) {
            Toast.makeText(context, "原来你叫"+game.name+"啊！", Toast.LENGTH_SHORT).show();
        }

        @Override
        public void onClickLike(int position, Game game) {
            Toast.makeText(context, "我也"+game.like+"这游戏！", Toast.LENGTH_SHORT).show();
        }
    }

    public class GameChildItem extends AssemblyChildItem<Game> {
        private ImageView iconImageView;
        private TextView nameTextView;
        private TextView likeTextView;

        public GameChildItem(int itemLayoutId, ViewGroup parent) {
            super(itemLayoutId, parent);
        }

        @Override
        protected void onFindViews(View itemView) {
            iconImageView = (ImageView) findViewById(R.id.image_gameListItem_icon);
            nameTextView = (TextView) findViewById(R.id.text_gameListItem_name);
            likeTextView = (TextView) findViewById(R.id.text_gameListItem_like);
        }

        @Override
        protected void onConfigViews(Context context) {
            iconImageView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    eventListener.onClickIcon(getChildPosition(), getData());
                }
            });
            nameTextView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    eventListener.onClickName(getChildPosition(), getData());
                }
            });
            likeTextView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    eventListener.onClickLike(getChildPosition(), getData());
                }
            });
        }

        @Override
        protected void onSetData(int groupPosition, int childPosition, boolean isLastChild, Game game) {
            iconImageView.setImageResource(game.iconResId);
            nameTextView.setText(game.name);
            likeTextView.setText(game.like);
        }
    }
}

File: sample/src/main/java/me/panpf/adapter/sample/itemfactory/GameChildItemFactory.kt
package me.panpf.adapter.sample.itemfactory

import android.content.Context
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import me.panpf.adapter.AssemblyItem
import me.panpf.adapter.AssemblyItemFactory
import me.panpf.adapter.ktx.bindView
import me.panpf.adapter.sample.R
import me.panpf.adapter.sample.bean.Game

class GameChildItemFactory(context: Context) : AssemblyItemFactory<Game>() {

    init {
        setOnViewClickListener(R.id.image_gameListItem_icon) { view, position, positionInPart, data ->
            Toast.makeText(context, "瞅这游戏这臭逼样！", Toast.LENGTH_SHORT).show()
        }
        setOnViewClickListener(R.id.text_gameListItem_name) { view, position, positionInPart, data ->
            Toast.makeText(context, "原来你叫" + data?.name + "啊！", Toast.LENGTH_SHORT).show()
        }
        setOnViewClickListener(R.id.text_gameListItem_like) { view, position, positionInPart, data ->
            Toast.makeText(context, "我也" + data?.like + "这游戏！", Toast.LENGTH_SHORT).show()
        }
    }

    override fun match(data: Any?): Boolean {
        return data is Game
    }

    override fun createAssemblyItem(parent: ViewGroup): GameChildItem {
        return GameChildItem(R.layout.list_item_game, parent)
    }

    inner class GameChildItem(itemLayoutId: Int, parent: ViewGroup) : AssemblyItem<Game>(itemLayoutId, parent) {
        private val iconImageView: ImageView by bindView(R.id.image_gameListItem_icon)
        private val nameTextView: TextView by bindView(R.id.text_gameListItem_name)
        private val likeTextView: TextView by bindView(R.id.text_gameListItem_like)

        override fun onConfigViews(context: Context) {
        }

        override fun onSetData(position: Int, game: Game?) {
            game ?: return
            iconImageView.setImageResource(game.iconResId)
            nameTextView.text = game.name
            likeTextView.text = game.like
        }
    }
}
